src/components/CustomTextInput.tsx

```diff
@@ -0,0 +1,118 @@
+  border: 1px solid ${props => props.borderColor};
+  padding: 0 ${props => props.theme.spacings.sm};
+  border-radius: 2px;
+  height: ${props => (props.lines > 1 ? props.lines * LINE_HEIGHT : MIN_HEIGHT)}px;
+  flex-direction: row;
+  margin-bottom: ${props => (props.showErrorValidation ? 0 : props.theme.spacings.xs)};
+`
+
+const IconContainer = styled.View<{ multiLine: boolean }>`
+  align-self: ${props => (props.multiLine ? 'flex-start' : 'center')};
+  padding: ${props => props.theme.spacings.xs} 0;
+`
+
+const ErrorContainer = styled.View`
+  min-height: ${props => props.theme.spacings.lg};
+`
+
+interface CustomTextInputProps extends TextInputProps {
+  value: string
+  onChangeText: (value: string) => void
+  lines?: number
+  clearable?: boolean
+  rightContainer?: ReactElement
+  errorMessage?: string
+  customBorderColor?: string
+}
+
+// depending on error and focus the border color will be defined
+const getBorderColor = (hasErrorMessage: boolean, isFocused: boolean): string => {
+  if (hasErrorMessage) {
+    return theme.colors.incorrect
+  }
+  return isFocused ? theme.colors.primary : theme.colors.textSecondary
+}
+
+const CustomTextInput = ({
+  value,
+  onChangeText,
+  lines = 1,
+  clearable = false,
+  placeholder,
+  textContentType,
+  rightContainer,
+  errorMessage,
+  editable = true,
+  onSubmitEditing,
+  customBorderColor
+}: CustomTextInputProps): ReactElement => {
+  const theme = useTheme()
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+  const hasErrorMessage = !!(errorMessage && errorMessage.length > 0)
+  const showErrorValidation = errorMessage !== undefined
+  const multiLine = lines > 1
+
+  return (
+    <>
+      <TextInputContainer
+        lines={lines}
+        borderColor={customBorderColor ?? getBorderColor(hasErrorMessage, isFocused)}
+        showErrorValidation={showErrorValidation}>
+        <StyledTextInput
+          onFocus={() => setIsFocused(true)}
+          onBlur={() => setIsFocused(false)}
+          textContentType={textContentType}
+          value={value}
+          onChangeText={onChangeText}
+          placeholder={placeholder}
+          multiline={multiLine}
+          placeholderTextColor={theme.colors.placeholder}
+          editable={editable}
+          onSubmitEditing={onSubmitEditing}
+          textAlignVertical={multiLine ? 'top' : 'center'}
+        />
+        <IconContainer multiLine={multiLine}>
+          {clearable && value.length > 0 ? (
+            <ClearContainer onPress={() => onChangeText('')} testID='clearInput'>
+              <CloseIcon width={wp('6%')} height={wp('6%')} />
+            </ClearContainer>
+          ) : (
+            rightContainer
+          )}
+        </IconContainer>
+      </TextInputContainer>
+      {showErrorValidation && (
+        <ErrorContainer>{errorMessage.length > 0 && <ContentError>{errorMessage}</ContentError>}</ErrorContainer>
```

Comment:
If you just use `hasErrorMessage` in the line above you don't have to check for `errorMessage.length > 0`

Reply:
that doesn't work because if you want to `showErrorValidation` you use the ErrorContainer as a placeholder even if there is currently no errorMessage. So you don't have jumping elements